test/object-store/util/sync/baas_admin_api.cpp

```diff
@@ -1081,14 +1081,8 @@ bool AdminAPISession::is_initial_sync_complete(const std::string& app_id) const
 {
     auto progress_endpoint = apps()[app_id]["sync"]["progress"];
     auto progress_result = progress_endpoint.get_json();
-    if (auto it = progress_result.find("progress"); it != progress_result.end() && it->is_object() && !it->empty()) {
-        for (auto& elem : *it) {
-            auto is_complete = elem["complete"];
-            if (!is_complete.is_boolean() || !is_complete.get<bool>()) {
-                return false;
-            }
-        }
-        return true;
+    if (auto it = progress_result.find("accepting_clients"); it != progress_result.end()) {
+        return it->get<bool>();
```

Comment:
@danieltabacaru I think this change may be causing races in the PBS client reset tests. See https://github.com/realm/realm-core/blob/master/test/object-store/util/sync/sync_test_utils.cpp#L543-L552
The previous code waits for all tables to be "complete" while the changed code now returns true much earlier in the bootstrapping process.

Reply:
@ironage i see. thanks for looking into it.
@tkaye407 is waiting for initial sync change supposed to work only for flx (assuming it's only a problem in pbs)?

Reply:
The purpose of this was to define when it is valid for clients to connect. The difference being that in PBS clients are allowed to connect immediately (while initial sync is running, which was honestly a pretty bad experience for things like client resets)

Reply:
@ironage the previous implementation seems to keep us out of trouble in tests, but IIUC the current implementation is how things work in production, so users could run into the same issue. I can revert the change for now

Reply:
@danieltabacaru  thanks. I don't think this is an issue for production, it is just about our test suite's expectation of the server state.